### PR TITLE
feat(teams): teams v1

### PR DESCRIPTION
**Pull request by samuelmasse**
Upgrades teams to be version 1.0.0. Added more content types and fixed some things. As for the API version nothing really changed for this channel. We were using botbuilder 4.14.0 before and now we use 4.15.0

Closes DEV-2277

### CODE REVIEW

File: packages/channels/package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "@slack/bolt": "^3.9.0",
     "axios": "^0.25.0",
+    "botbuilder": "^4.15.0",
     "cli-color": "^2.0.1",
```
Comment (samuelmasse) [inline packages/channels/package.json:31]: We've upgraded a whopping 1 minor version here (from 4.14.0). I think I may have updated this channel before.

File: packages/channels/src/teams/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+| -------- | :---: |
+| Text     |  ✅   |
+| Image    |  ✅   |
+| Choice   |  ✅   |
+| Dropdown |  ✅   |
+| Card     |  ✅   |
+| Carousel |  ✅   |
+| File     |  ✅   |
+| Audio    |  ✅   |
+| Video    |  ✅   |
```
Comment (samuelmasse) [inline packages/channels/src/teams/README.md:13]: We send these as text with a link

File: packages/channels/src/teams/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+| Location |  ✅   |
+
+### Receiving
+
+| Channels      | Teams |
+| ------------- | :---: |
+| Text          |  ✅   |
+| Quick Reply   |  ✅   |
+| Postback      |  ✅   |
+| Say Something |  ✅   |
```
Comment (samuelmasse) [inline packages/channels/src/teams/README.md:23]: I sorted these out. Turns out this channel only received text, even if it was quick reply, postback and say something. Now all three work.

File: packages/channels/src/teams/config.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import Joi from 'joi'
+import { ChannelConfig } from '../base/config'
+
+export interface TeamsConfig extends ChannelConfig {
+  appId: string
+  appPassword: string
+}
```
Comment (samuelmasse) [inline packages/channels/src/teams/config.ts:7]: Removed the tenantId here because I don't know what it's used for. We can add it back if we need it.

Comment (laurentlp) [reply packages/channels/src/teams/config.ts:7]: I think it was for some validation but I'm not sure either.

**Review by laurentlp — APPROVED** [review]

**Pull request merged**